lib/srv/db/cloud/watchers/watcher.go

```diff
@@ -172,6 +194,55 @@ func makeFetchers(clients cloud.Clients, matchers []services.AWSMatcher) (result
 	return result, nil
 }
 
+func makeAzureFetchers(ctx context.Context, clients cloud.Clients, matchers []services.AzureMatcher) (result []Fetcher, err error) {
+	subIDsClient, err := clients.GetAzureSubscriptionIDsClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	for _, matcher := range matchers {
+		reduceAzureMatcher(&matcher)
+		subIDs := matcher.Subscriptions
+		if utils.SliceContainsStr(subIDs, types.Wildcard) {
+			// hit the subscriptions API at most once
+			subIDs, err = subIDsClient.ListSubscriptionIDs(ctx, common.MaxPages, true /*useCache*/)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
```

Comment:
Can we use a variable to save the subIDs returned from `subIDsClient.ListSubscriptionIDs` instead of using internal client cache that requires additional param  ?

Reply:
ok